src/v/cluster/rm_stm.cc

```diff
@@ -2755,112 +2646,6 @@ std::ostream& operator<<(std::ostream& o, const rm_stm::abort_snapshot& as) {
     return o;
 }
 
-void rm_stm::spawn_background_clean_for_pids(rm_stm::clear_type type) {
-    switch (type) {
-    case rm_stm::clear_type::tx_pids: {
-        if (
-          _log_state.fence_pid_epoch.size() <= _max_concurrent_producer_ids()) {
-            return;
-        }
-        break;
-    }
-    case rm_stm::clear_type::idempotent_pids: {
-        if (
-          _log_state.lru_idempotent_pids.size()
-          <= _max_concurrent_producer_ids()) {
-            return;
-        }
-        break;
-    }
-    }
-
-    ssx::spawn_with_gate(_gate, [this, type] { return clear_old_pids(type); });
-}
-
-ss::future<> rm_stm::clear_old_pids(clear_type type) {
-    auto try_lock = _clean_old_pids_mtx.try_get_units();
-    if (!try_lock) {
-        co_return;
-    }
-
-    auto read_lock = co_await _state_lock.hold_read_lock();
-
-    switch (type) {
-    case rm_stm::clear_type::tx_pids: {
-        co_return co_await clear_old_tx_pids();
-    }
-    case rm_stm::clear_type::idempotent_pids: {
-        co_return co_await clear_old_idempotent_pids();
-    }
-    }
-}
-
-ss::future<> rm_stm::clear_old_tx_pids() {
```

Comment:
this looks worrisome, you mentioned that this is mostly idempotency change only, but here we remove tx clearing logic; is it reintroduced later in the PR?

Reply:
yep it looks like an overlook, `lru_idempotent_pids` isn't filled for tx pids so we can't reply on producer state manager to drive old tx clean up